src/twisted/conch/interfaces.py

```diff
@@ -113,6 +113,35 @@ def closed():
 
 
 
+class ISessionSetEnv(Interface):
+    """A session that can set environment variables."""
+
+    def setEnv(name, value):
+        """
+        Set an environment variable for the shell or command to be started.
+
+        From U{RFC 4254, section 6.4
+        <https://tools.ietf.org/html/rfc4254#section-6.4>}: "Uncontrolled
+        setting of environment variables in a privileged process can be a
+        security hazard.  It is recommended that implementations either
+        maintain a list of allowable variable names or only set environment
+        variables after the server process has dropped sufficient
+        privileges."
+
+        (OpenSSH refuses all environment variables by default, but has an
+        C{AcceptEnv} configuration option to select specific variables to
+        accept.)
+
+        @param name: The name of the environment variable to set.
+        @type name: L{bytes}
+        @param value: The value of the environment variable to set.
+        @type value: L{bytes}
+        @raise ValueError: if setting this environment variable is not
```

Comment:
`ValueError` seems like a pretty broad exception to catch with no logging at all.  For example, if your application code is expecting an `int`-able value and your client transmitted `"1.0"`, you'll get a `ValueError` and it'll get swallowed.

In general, this is why using stdlib exceptions is generally a bad idea for specific failure modes.

Could we define a new exception here, `EnvironmentVariableNotPermitted`, so it's clear that it's being raised intentionally?  Alternately, if the only two possible behaviors here are "variable was set, OK" and "variable rejected", perhaps the return value ought to be an enum or something similar, and exceptions shouldn't be allowed in the interface?